bundles/mapping/mapmodule/plugin/layers/LayerSelectionPlugin.js

```diff
@@ -19,32 +19,44 @@ Oskari.clazz.define('Oskari.mapframework.bundle.mapmodule.plugin.LayerSelectionP
         me._defaultLocation = 'top left';
         me._index = 70;
         me._name = 'LayerSelectionPlugin';
+        me._toolOpen = false;
 
         me.initialSetup = true;
         me.templates = {};
         me._mobileDefs = {
-            buttons:  {
+            buttons: {
                 'mobile-layerselection': {
                     iconCls: 'mobile-layers',
                     tooltip: '',
                     sticky: true,
                     show: true,
                     callback: function () {
-                        if (me.popup && me.popup.isVisible()) {
-                            var sandbox = me.getSandbox();
-                            sandbox.postRequestByName('Toolbar.SelectToolButtonRequest', [null, 'mobileToolbar-mobile-toolbar']);
-                            me.popup.close(true);
-                            me.popup = null;
-                        } else {
-                            me.openSelection(true);
-                        }
+                        me._toggleToolState();
                     },
                     toggleChangeIcon: true
                 }
             },
             buttonGroup: 'mobile-toolbar'
         };
     }, {
+        _toggleToolState: function () {
+            var el = this.getElement();
+
+            if (this._toolOpen) {
+                if (el) {
+                    el.removeClass('active');
+                }
+                this._toolOpen = false;
```

Comment:
Isn't the popup.isVisible() good enough indication about this? Do we really want another variable for tracking the state?